src/modbus.cpp

```diff
@@ -765,21 +799,21 @@ void modbus::ParseData() {
       } else if (datatype == "string") {
         //********** handle Datatype String ***********//
         if (!posArray.isNull()) { 
-          char buffer[posArray.size()];
+          char buf[posArray.size()+1];        // set up c-string to hold all bytes and the trailing 0x00-char
```

Comment:
warum buf und nicht buffer?
ist aber schon in der aktuellen dev-branch behoben

Reply:
ich habe mal gelernt, daß unterschiedliche Variablen auch verschiedene Namen haben sollten, um die Übersicht zu behalten; und je kleiner der Lebensbereich, desto kürzer darf der Name sein - also buf.

Da hat aber jeder seine Philosophie, ich hatte wohl auch hier zeitgleich mit Dir die Lösung eingebracht...